python/idsse_common/idsse/common/vectaster.py

```diff
@@ -0,0 +1,254 @@
+
+    Raises:
+        ValueError: If geometry type is not currently supported
+
+    Returns:
+        Geometry: Shapely geometry of the same type as input geometry with vertices
+                    defined by x,y pixels
+    """
+    if isinstance(geo, LineString):
+        return geographic_linestring_to_pixel(geo, grid_proj, rounding)
+    if isinstance(geo, Polygon):
+        return geographic_polygon_to_pixel(geo, grid_proj, rounding)
+    raise ValueError('Passed geometry is not of supported types')
+
+
+def geographic_linestring_to_pixel(
+        line_string: LineString,
+        grid_proj: GridProj,
+        rounding: RoundingMethod = None
+) -> LineString:
+    """Map a LineString specified in lat/lon space to geometry specified in pixel space
+
+    Args:
+        line_string (LineString): Shapely geometry with vertices defined by lon,lat
+        grid_proj (GridProj): Projection plus pixel resolution
+        rounding (RoundingMethod, optional): One of None, 'floor', 'round'. Defaults to None.
+
+    Raises:
+        ValueError: If geometry is not a LineString
+    Returns:
+        LineString: Shapely LineString with vertices defined by x,y pixels
+    """
+    if not isinstance(line_string, LineString):
+        raise ValueError(f'Geometry must be a LineString but is a {type(line_string)}')
+
+    coords = [grid_proj.map_geo_to_pixel(*ll, rounding) for ll in line_string.coords]
+
+    pixel_linestring = LineString(coords)
+    return pixel_linestring
+
+
+def geographic_polygon_to_pixel(
+        poly: Polygon,
+        grid_proj: GridProj,
+        rounding: RoundingMethod = None
+) -> Polygon:
+    """Map a Polygon specified in lat/lon space to geometry specified in pixel space
+
+    Args:
+        poly (Polygon): Shapely geometry with vertices defined by lon,lat
+        grid_proj (GridProj): Projection plus pixel resolution
+        rounding (RoundingMethod, optional): One of None, 'floor', 'round'. Defaults to None.
+
+    Raises:
+        ValueError: If geometry is not a LineString
+    Returns:
+        Polygon: Shapely Polygon with vertices defined by x,y pixels
+    """
+    if not isinstance(poly, Polygon):
+        raise ValueError(f'Geometry must be a Polygon but is a {type(poly)}')
+
+    print(grid_proj)
+    exterior = [grid_proj.map_geo_to_pixel(*ll, rounding) for ll in poly.exterior.coords]
+    interiors = [[grid_proj.map_geo_to_pixel(*ll, rounding) for ll in interior.coords]
+                 for interior in poly.interiors]
+
+    pixel_poly = Polygon(exterior, holes=interiors)
+    return pixel_poly
+
+
+def pixels_near_line_string(line_string: LineString,
+                            dist: float = 0.25) -> Tuple[numpy.array]:
+    """Determine the pixels that are close to a specified LineString
+
+    Args:
+        line_string (LineString): Shapely geometry with vertices defined by x,y
+        dist (float, optional): Distance cutoff used to determine in a pixel is "near".
```

Comment:
The distance is in whatever space line_string is in, most likely pixel. Though technically it could be lat/lon if someone is using a lat/lon grid, not normal but I've needed that in the past.

Good catch on the distance and min/max, for the code to work correctly the distance needs to be added/subtracted before floor/ceiling. If I leave the currently implementation I'll include that, but I think I might rewrite the core logic because I don't think this code actually gets the results that I want.